--- zmailer/proto/cf/aliases.cf.orig	2004-07-11 18:10:34.000000000 +0200
+++ zmailer/proto/cf/aliases.cf	2004-08-08 11:32:53.000000000 +0200
@@ -620,11 +620,11 @@
 
 	(.+)-owner
 		listname="\1"
-		mailman_todo="mailowner"
+		mailman_todo="owner"
 		;;
 	(.+)-admin
 		listname="\1"
-		mailman_todo="mailowner"
+		mailman_todo="owner"
 		;;
 	(.+)-bounces
 		listname="\1"
@@ -644,7 +644,7 @@
 		;;
 	(.+)-request
 		listname="\1"
-		mailman_todo="mailcmd"
+		mailman_todo="request"
 		;;
 	(.+)-subscribe
 		listname="\1"
@@ -677,7 +677,7 @@
 	if [ -d "$MAILMAN_ROOT/lists/$listname" ] ; then
 		nattr=$(newattribute $attr privilege "0")
 		$(zapDSNnotify $nattr delivered "$sender" "$lcuser$domain")
-		a=(((local pipe.$listname "|$MAILMAN_ROOT/mail/wrapper $mailman_todo $listname" $nattr)))
+		a=(((local pipe.$listname "|$MAILMAN_ROOT/mail/mailman $mailman_todo $listname" $nattr)))
 		postzapDSNnotify a +
 		return $a
 	fi
