--- zmailer/configure.in.orig	2004-04-05 05:30:53.000000000 +0200
+++ zmailer/configure.in	2004-04-26 19:54:45.000000000 +0200
@@ -2204,8 +2204,8 @@
             CPPFLAGS="$CPPFLAGS $INCLSPF"
             AC_TRY_LINK([#include <sys/types.h>
 			#include <spf_alt/spf.h>
-                        int major, minor;],
-			[SPF_get_lib_version(&major,&minor);],
+                        int major, minor, patch;],
+			[SPF_get_lib_version(&major,&minor,&patch);],
                         ac_cv_lib_spf=yes,ac_cv_lib_spf=no)
 dnl AC_MSG_RESULT([ ... link -lspf yields: $ac_cv_lib_spf])
             if test "$ac_cv_lib_spf" = no; then
@@ -2213,8 +2213,8 @@
                 LIBS="$LIBS $LIBSOCKET -L/usr/local/lib $LIBSPF"
                 AC_TRY_LINK([#include <sys/types.h>
 				#include <spf_alt/spf.h>
-				int major, minor;],
-				[SPF_get_lib_version(&major,&minor);],
+				int major, minor, patch;],
+				[SPF_get_lib_version(&major,&minor,&patch);],
                         ac_cv_lib_spf=yes,ac_cv_lib_spf=no)
 dnl AC_MSG_RESULT([ ... link -L/usr/local/lib -lspf yields: $ac_cv_lib_spf])
                 if test  "$ac_cv_lib_spf" = yes; then
@@ -2227,8 +2227,8 @@
                 LIBS="$LIBS $LIBSOCKET -L/local/lib $LIBSPF"
                 AC_TRY_LINK([#include <sys/types.h>
 				#include <spf_alt/spf.h>
-				int major, minor;],
-				[SPF_get_lib_version(&major,&minor);],
+				int major, minor, patch;],
+				[SPF_get_lib_version(&major,&minor,&patch);],
                         ac_cv_lib_spf=yes,ac_cv_lib_spf=no)
 dnl AC_MSG_RESULT([ ... link -L/local/lib -lspf yields: $ac_cv_lib_spf])
                 if test  "$ac_cv_lib_spf" = yes; then
@@ -2241,8 +2241,8 @@
                 LIBS="$LIBS $LIBSOCKET -L/opt/lib $LIBSPF"
                 AC_TRY_LINK([#include <sys/types.h>
 				#include <spf_alt/spf.h>
-				int major, minor;],
-				[SPF_get_lib_version(&major,&minor);],
+				int major, minor, patch;],
+				[SPF_get_lib_version(&major,&minor,&patch);],
                         ac_cv_lib_spf=yes,ac_cv_lib_spf=no)
 dnl AC_MSG_RESULT([ ... link -L/opt/lib -lspf yields: $ac_cv_lib_spf])
                 if test  "$ac_cv_lib_spf" = yes; then
@@ -2255,8 +2255,8 @@
                 LIBS="$LIBS $LIBSOCKET -L/opt/local/lib $LIBSPF"
                 AC_TRY_LINK([#include <sys/types.h>
 				#include <spf_alt/spf.h>
-				int major, minor;],
-				[SPF_get_lib_version(&major,&minor);],
+				int major, minor, patch;],
+				[SPF_get_lib_version(&major,&minor,&patch);],
                         ac_cv_lib_spf=yes,ac_cv_lib_spf=no)
                 if test  "$ac_cv_lib_spf" = yes; then
                     spfloc="-L/opt/local/lib"
@@ -2300,8 +2300,8 @@
         AC_CACHE_VAL(ac_cv_lib_spf,
           [AC_TRY_LINK([#include <sys/types.h>
 			#include <spf_alt/spf.h>
-			int major, minor;],
-			[SPF_get_lib_version(&major,&minor);],
+			int major, minor, patch;],
+			[SPF_get_lib_version(&major,&minor,&patch);],
                         ac_cv_lib_spf=yes,ac_cv_lib_spf=no)
           ])
         if test "$ac_cv_lib_spf" = yes; then
