--- router/libdb/ldap.c.orig	2004-01-31 04:21:14.174029408 +0000
+++ router/libdb/ldap.c	2004-01-31 04:21:28.411864928 +0000
@@ -59,7 +59,7 @@
 	if (lmap->ld != NULL) {
 
 		if (lmap->protocol > 0 &&
-		    ldap_set_option( ldap->ld,
+		    ldap_set_option( lmap->ld,
 				     LDAP_OPT_PROTOCOL_VERSION,
 				     &lmap->protocol ) != LDAP_OPT_SUCCESS ) {
 		  fprintf( stderr, "Could not set LDAP_OPT_PROTOCOL_VERSION %d\n", lmap->protocol );
